sympy/tensor/array/arrayop.py

```diff
@@ -208,3 +208,38 @@ def derive_by_array(expr, dx):
             return ImmutableDenseNDimArray([expr.diff(i) for i in dx], dx.shape)
         else:
             return diff(expr, dx)
+
+
+def permutedims(expr, perm):
+    """
+    Permutes the indices of an array.
+
+    Parameter specifies the permutation of the indices.
+    """
+    if not isinstance(expr, NDimArray):
+        raise TypeError("expression has to be an N-dim array")
+
+    perm = tuple(perm)
```

Comment:
I don't want to make it too complicated. Experience has shown that end-users get confused if they have to use too many classes.